server/src/test/java/org/elasticsearch/rest/action/admin/indices/RestValidateQueryActionTests.java

```diff
@@ -46,21 +49,21 @@
 
 public class RestValidateQueryActionTests extends AbstractSearchTestCase {
 
-    private static ThreadPool threadPool = new TestThreadPool(RestValidateQueryActionTests.class.getName());
-    private static NodeClient client = new NodeClient(Settings.EMPTY, threadPool);
+    private ThreadPool threadPool = new TestThreadPool(RestValidateQueryActionTests.class.getName());
```

Comment:
I had to change the scope of threadPool and others otherwise the threadpool.ThreadContext is not correctly cleaned